src/main/java/com/microsoft/hdinsight/utils/AzureUtils.java

```diff
@@ -199,12 +210,46 @@ private static StorageAccount fetchStorageAccount(HWCClusterConfig clusterConfig
 
   private static OsProfile addSSHCredentials(HWCClusterConfig clusterConfig) {
     LOG.info("Setting up SSH credentials...");
+    if (clusterConfig.getClusterCredentials().getSshCredentials() instanceof SSHWithPassword) {
+      return addSSHWithPasswordCredentials((SSHWithPassword) clusterConfig.getClusterCredentials().getSshCredentials());
+    } else {
+      return addSSHWithKeysCredentials((SSHWithKeys) clusterConfig.getClusterCredentials().getSshCredentials());
+    }
+  }
+
+  private static OsProfile addSSHWithKeysCredentials(SSHWithKeys sshWithKeys) {

```

Comment:
Test what happens for invalid ssh key

Reply:
Modified the code to throw an exception if the ssh key path is non-existent or a directory